src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2

```diff
@@ -400,7 +400,11 @@ REPORT_CACHE = 'redis'
 redis_cache = {
     'BACKEND': 'django_redis.cache.RedisCache',
 {% if 'REDIS_DB' in localsettings %}
+{% if localsettings.get('REDIS_PASSWORD') %}
+    'LOCATION': 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}/{{ localsettings.REDIS_DB }}',
+{% else %}
     'LOCATION': 'redis://{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}/{{ localsettings.REDIS_DB }}',
+{% endif %}
 {% else %}
     'LOCATION': 'redis://{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}',
 {% endif %}
```

Comment:
How about this change so we can support password even if you don't specify a database? Also it prevents having nested if's

```suggestion
{% set redis_url = 'redis://:{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
{% if localsettings.get('REDIS_PASSWORD') %}
{% set redis_url = 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
{% endif %}
{% if 'REDIS_DB' in localsettings %}
    'LOCATION': '{{ redis_url }}/{{ localsettings.REDIS_DB }}',
{% else %}
    'LOCATION': '{{ redis_url }}',
{% endif %}
```